### PR TITLE
Add initial test for RTCDTMFSender

### PR DESCRIPTION
**Pull request by soareschen**
This adds test for the final uncovered section in webrtc-pc.

This test makes the assumption discussed in w3c/webrtc-pc#1415, that the `dtmf` attribute for `RTCRtpSender` is always set for audio transceivers. This is required as otherwise it is impossible to reliably obtain an instance of RTCDTMFSender.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/259316793)

# Firefox (nightly) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/259316794)

# Sauce (safari) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/259316800)

# Chrome (unstable) #

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/259316801)

# Sauce (MicrosoftEdge) #

**Review by rwaldron — CHANGES_REQUESTED** [review]

File: webrtc/RTCPeerConnection-helper.js [hunk 1]
```diff
@@ -384,6 +384,7 @@ function getTrackFromUserMedia(kind) {
     const tracks = mediaStream.getTracks();
     assert_greater_than(tracks.length, 0,
       `Expect getUserMedia to return at least one track of kind ${kind}`);
+
```
Comment (soareschen) [inline webrtc/RTCPeerConnection-helper.js]: Missed your comment. Have fixed it now. Hope the CI will pass now.

**Review by rwaldron — CHANGES_REQUESTED** [review]
Minor fixes needed

File: webrtc/RTCDTMFSender-ontonechange.html [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+    7.2.  insertDTMF
+      11. If a Playout task is scheduled to be run; abort these steps; otherwise queue
+          a task that runs the following steps (Playout task):
+        3.  If toneBuffer is an empty string, fire an event named tonechange with an
+            empty string at the RTCDTMFSender object and abort these steps.
+        4.  Remove the first character from toneBuffer and let that character be tone.
+        5.  Queue a task to be executed in duration + interToneGap ms from now that
+            runs the steps labelled Playout task.
+        6.  Fire an event named tonechange with a string consisting of tone at the
+            RTCDTMFSender object.
```
Comment (soareschen) [inline webrtc/RTCDTMFSender-ontonechange.html]: Nice catch! I have fixed them now.

Step 5 is excluded because it is untestable.

Comment (rwaldron) [inline webrtc/RTCDTMFSender-ontonechange.html]: Got it!

**Comment by rwaldron** [comment]
I tried to add you as a CC on this bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1290948 but bugzilla could not find your email address.

**Review by rwaldron — APPROVED** [review]

**Comment by soareschen** [comment]
> I tried to add you as a CC on this bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1290948 but bugzilla could not find your email address.

Hmm I do have an account on Bugzilla. Did you use my main email soares.chen@gmail.com?

**Pull request merged**